src/Mod/Path/App/Area.cpp

```diff
@@ -332,60 +332,31 @@ int Area::addShape(CArea &area, const TopoDS_Shape &shape, const gp_Trsf *trsf,
 static std::vector<gp_Pnt> discretize(const TopoDS_Edge &edge, double deflection) {
     std::vector<gp_Pnt> ret;
     BRepAdaptor_Curve curve(edge);
-    Standard_Real efirst,elast,first,last;
+    Standard_Real efirst,elast;
     efirst = curve.FirstParameter();
     elast = curve.LastParameter();
     bool reversed = (edge.Orientation()==TopAbs_REVERSED);
 
     // push the first point
     ret.push_back(curve.Value(reversed?elast:efirst));
 
-    Handle(Geom_Curve) c = BRep_Tool::Curve(edge, first, last);
-    first = c->FirstParameter();
-    last = c->LastParameter();
-    if(efirst>elast) {
-        if(first<last)
-            std::swap(first,last);
-    }else if(first>last)
-        std::swap(first,last);
-
-    // NOTE: OCCT QuasiUniformDeflection has a bug cause it to return only
-    // partial points for some (BSpline) curve if we pass in the edge trimmed
-    // first and last parameters. Passing the original curve first and last
-    // parameters works fine. The following algorithm uses the original curve
-    // parameters, and skip those out of range. The algorithm shall work the
-    // same for any other discetization algorithm, althgouth it seems only
-    // QuasiUniformDeflection has this bug.
-
     // NOTE: QuasiUniformDeflection has trouble with some B-Spline, see
     // https://forum.freecadweb.org/viewtopic.php?f=15&t=42628
     //
     // GCPnts_QuasiUniformDeflection discretizer(curve, deflection, first, last);
     //
-    GCPnts_UniformDeflection discretizer(curve, deflection, first, last);
+    GCPnts_UniformDeflection discretizer(curve, deflection, efirst, elast);
     if (!discretizer.IsDone ())
         Standard_Failure::Raise("Curve discretization failed");
     if(discretizer.NbPoints () > 1) {
         int nbPoints = discretizer.NbPoints ();
         //strangely OCC discretizer points are one-based, not zero-based, why?
         if(reversed) {
             for (int i=nbPoints-1; i>=1; --i) {
-                auto param = discretizer.Parameter(i);
-                if(first<last) {
-                    if(param<efirst || param>elast)
-                        continue;
-                }else if(param>efirst || param<elast)
-                    continue;
```

Comment:
Since we aren't passing in out-of range first / last parameters any more, the skipping logic here is no longer needed.